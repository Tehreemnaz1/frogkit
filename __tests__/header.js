import { mount } from 'vue-test-utils'
import Vue from 'vue'
import Header from '@/components/header'
import countries from '../data/countries'

const navMainItems = [
  { label: 'Kamera', url: 'https://www.smartfrog.com/de-de/kamera/' },
  { label: 'Videoaufnahme', url: 'https://www.smartfrog.com/de-de/videoaufnahme/' },
  { label: 'Einsatzbereiche', url: 'https://www.smartfrog.com/de-de/einsatzbereiche/' },
  { label: 'So geht\'s', url: 'https://www.smartfrog.com/de-de/so-gehts/' }
]
const navExtraItems = [
  { label: 'Login', url: 'https://app.smartfrog.com/de-de/login' },
  { label: 'Shop', url: 'https://www.smartfrog.com/de-de/shop/products' },
  { label: 'Warenkorb', url: 'https://www.smartfrog.com/de-de/shop/cart' }
]
const formItems = {
  emailPlaceholder: 'E-Mail Adresse',
  passwordPlaceholder: 'Passwort',
  registerLabel: 'Noch kein Benutzerkonto?',
  registerUrl: 'https://www.smartfrog.com//de-de/shop/register',
  retriveLabel: 'Passwort vergessen?',
  retriveUrl: '/de-de/shop/resetpassword'
}

const testCountry = {
  "countryCode": "IO",
  "countryName": "British Indian Ocean Territory",
  "market": "NHW_2",
  "currency": "USD",
  "currencySign": "US$",
  "languages": {
    "en":"British Indian Ocean Territory"
  },
  "preferred": false
}

const data = {
  navMainItems: navMainItems,
  navExtraItems: navExtraItems,
  cartCounter: 1,
  formItems: formItems,
  countries: countries,
  currentCountry: 'de',
  countrySelectLabel: 'Your Country is missing? Click here:',
  countrySelectPlaceholder: 'Please select a country'
}

describe('Header component', () => {
  it('Should render component and match snapshot', () => {
    const wrapper = mount(Header, { propsData: data })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Login button should emit login event when clicked', () => {
    const wrapper = mount(Header, { propsData: data })
    wrapper.vm.$emit('login')
    expect(wrapper.emitted().login).toBeTruthy()
  })

  it('Login button should emit email and password values when clicked', () => {
    const wrapper = mount(Header, { propsData: data })
    wrapper.vm.$emit('login')
    wrapper.vm.$emit('login', 'test@test.com', 'test-password')
    expect(wrapper.emitted().login.length).toBe(2)
    expect(wrapper.emitted().login[1]).toEqual(['test@test.com', 'test-password'])
  })

  it('Should emit countryChange event when country is changed in Country Selector child component', () => {
    const wrapper = mount(Header, { propsData: data })
    wrapper.vm.$emit('countryChange')
    expect(wrapper.emitted().countryChange).toBeTruthy()
  })

  it('Should emit the selected country object when country is changed in Country Selector child component', () => {
    const wrapper = mount(Header, { propsData: data })
    wrapper.vm.$emit('countryChange')
    wrapper.vm.$emit('countryChange', testCountry)
    expect(wrapper.emitted().countryChange.length).toBe(2)
    expect(wrapper.emitted().countryChange[1]).toEqual([testCountry])
  })
})
