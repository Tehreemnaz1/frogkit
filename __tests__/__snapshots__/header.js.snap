// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render component and match snapshot 1`] = `
<header class="fk-header">
    <div class="fk-header__container">
        <h1 class="fk-header__logo"><a href="/"><img src="../assets/logo.svg"></a></h1> <span class="fk-header__toggle"><icon icon="hamburger-menu"></icon></span>
        <nav class="fk-header__nav">
            <ul>
                <li><a href="https://www.smartfrog.com/de-de/kamera/">Kamera</a></li>
                <li><a href="https://www.smartfrog.com/de-de/videoaufnahme/">Videoaufnahme</a></li>
                <li><a href="https://www.smartfrog.com/de-de/einsatzbereiche/">Einsatzbereiche</a></li>
                <li><a href="https://www.smartfrog.com/de-de/so-gehts/">So geht's</a></li>
                <li><a href="https://www.smartfrog.com/de-de/shop/cart">Warenkorb</a></li>
                <li><a href="https://www.smartfrog.com/de-de/shop/products">Shop</a></li>
                <li><a href="https://app.smartfrog.com/de-de/login">Login</a></li>
            </ul>
        </nav>
        <flex grow="" justify="end" align="center">
            <countryselector countries="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
                currentcountry="de" selectlabel="Your Country is missing? Click here:"></countryselector>
            <div class="fk-header__form">
                <div class="fk-header__form__field"><input name="email" type="email" size="small" placeholder="E-Mail Adresse"> <a href="https://www.smartfrog.com//de-de/shop/register">Noch kein Benutzerkonto?</a></div>
                <div class="fk-header__form__field"><input name="password" type="password" size="small" placeholder="Passwort"> <a href="/de-de/shop/resetpassword">Passwort vergessen?</a></div>
            </div> <button link="" color="primary" size="small" class="fk-header__btn-login">
        Login
      </button> <button link="" color="secondary" size="small" href="https://www.smartfrog.com/de-de/shop/products" class="fk-header__btn-shop">
        Shop
      </button> <button link="" color="secondary" size="small" href="https://www.smartfrog.com/de-de/shop/cart" class="fk-header__btn-cart"><icon icon="basket"></icon> <span class="fk-header__btn-cart__counter">1</span></button></flex>
    </div>
</header>
`;
